### PR TITLE
Activate GitHub Actions on all PRs and on push

### PR DESCRIPTION
**Pull request by eolivelli**
### Motivation

Currently we are not running PR validation for Pull Requests against release branches (as branch-4.12).
Also we are not running tests on master branch and we do not have Jenkins anymore

### Changes
- Enable PR validation for pull requests against all release branches
- Enable GitHub actions on "push"

### CODE REVIEW

**Comment by eolivelli** [comment]
@lhotari can you please take a look ?

File: .github/workflows/compatibility-check-java8.yml [hunk 1]
```diff
@@ -20,9 +20,11 @@
 name: Compatibility Check Java8
 
 on:
+  push;
```
Comment (lhotari) [inline .github/workflows/compatibility-check-java8.yml:23]: looks like a typo. `;` -> `:`

**Review by lhotari — CHANGES_REQUESTED** [review]

File: .github/workflows/compatibility-check-java8.yml (hunk 1 above)
Comment (eolivelli) [reply .github/workflows/compatibility-check-java8.yml]: thanks @lhotari 
fixed

**Review by lhotari — APPROVED** [review]

**Comment by eolivelli** [comment]
@sijie can you please help merging this patch ?

**Review by Ghatage — APPROVED** [review]
LGTM +1

**Comment by Ghatage** [comment]
Big man might be busy, I gotchu @eolivelli

**Pull request merged**